sentry-android-replay/src/main/java/io/sentry/android/replay/ReplayCache.kt

```diff
@@ -237,9 +240,169 @@ public class ReplayCache internal constructor(
             encoder?.release()
             encoder = null
         }
+        isClosed.set(true)
+    }
+
+    // TODO: it's awful, choose a better serialization format
```

Comment:
it's pretty bad right now, cause we load the entire thing into memory, then modify, then serialize it back, and this is called quite often (especially when there's touch events going on). If you have better ideas here, please throw them in.

My one idea so far was to use something like `RandomAccessFile` and only read/write at the specified position directly from/to file

Reply:
how big is the full map?
is it feasible to keep it in memory and save to disk on close()?

Reply:
it's actually small 5-10 entries. I'll measure the perf currently, maybe it's not that bad

> is it feasible to keep it in memory and save to disk on close()?

Unfortunately, no, because we don't know when the process is terminated (and close() is not called in that case), so we have to continuously persist the values to have the latest state on restart

Reply:
okie, I've made a small improvement for now where we:
* Don't read from the file, but keep the map in-memory (only read on the first access)
* Moved file instantiation out and made it lazy
* Write the in-memory map on change

I'm still going to rework it and use RandomAccessFile most likely (and probably flush periodically, let's see), but a bit later, for now I've measured the time and it's taking 0.5ms on average to write the entire map (also done on a background thread), so I think it's ok for the first iteration